### PR TITLE
curve tricrypto-ng: pool tracker

### PR DESCRIPTION
**Pull request by it4rb**

### CODE REVIEW

**Review by hanguyenk — APPROVED** [review]

File: pkg/liquidity-source/curve/tricrypto-ng/pool_tracker.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	if err != nil {
+		lg.WithFields(logger.Fields{"error": err}).Error("failed to marshal extra data")
+		return entity.Pool{}, err
+	}
+
+	var reserves = make(entity.PoolReserves, 0, len(balances)+1)
+	for i := range balances {
+		reserves = append(reserves, balances[i].String())
+	}
+
```
Comment (hieuvo) [inline pkg/liquidity-source/curve/tricrypto-ng/pool_tracker.go:236]: don't we append `lpSupply` to `reserves`?

File: pkg/liquidity-source/curve/tricrypto-ng/pool_tracker.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+		MidFee:              number.SetFromBig(midFee),
+		OutFee:              number.SetFromBig(outFee),
+		LpSupply:            number.SetFromBig(lpSupply),
+		XcpProfit:           number.SetFromBig(xcpProfit),
+		VirtualPrice:        number.SetFromBig(virtualPrice),
+		AllowedExtraProfit:  number.SetFromBig(allowedExtraProfit),
+		AdjustmentStep:      number.SetFromBig(adjustmentStep),
+	}
+	extra.PriceScale = make([]uint256.Int, len(priceScales))
+	lo.ForEach(priceScales, func(item *big.Int, i int) { extra.PriceScale[i].SetFromBig(item) })
```
Comment (hieuvo) [inline pkg/liquidity-source/curve/tricrypto-ng/pool_tracker.go:220]: just wonder why we have to use `lo.ForEach`, while a simple for loop works well here?

File: pkg/liquidity-source/curve/tricrypto-ng/pool_tracker.go (hunk 1 above)
Comment (it4rb) [reply pkg/liquidity-source/curve/tricrypto-ng/pool_tracker.go:236]: um no, we have it in `extra` already, I don't really know why we put that in to reserves previously

File: pkg/liquidity-source/curve/tricrypto-ng/pool_tracker.go (hunk 2 above)
Comment (it4rb) [reply pkg/liquidity-source/curve/tricrypto-ng/pool_tracker.go:220]: um no specific reason, I just find them more familiar to me (coming from Scala background) and less code than the verbose loop

I've been bitten by some bugs in handwritten loop before, for example this code to split chunk (https://github.com/KyberNetwork/limit-order/pull/223/files#diff-d790a627480c897fa30fb5220132d0c44c4ef84fcc498155633d10ffe71de059R287). I know the bug is simple and could have been avoided. But I think if we had used `lo` at the beginning then we don't even have to think about that

**Pull request merged**